sycl/include/sycl/builtins.hpp

```diff
@@ -1377,6 +1377,198 @@ detail::common_rel_ret_t<T> signbit(T x) __NOEXC {
       __sycl_std::__invoke_SignBitSet<detail::internal_rel_ret_t<T>>(x));
 }
 
+// marray relational functions
+template <typename T,
+          typename = detail::enable_if_t<
+              detail::is_mgenfloat<T>::value &&
+                  detail::is_svgenfloat<detail::marray_element_type<T>>::value,
```

Comment:
Can we update definition of `is_mgenfloat` to automatically check if type is `float`, `double` or `half`? I just want this generic name definition to be closer to one we have in the spec. And it will also simplify the code

Reply:
Will add support of `marray<double,{N}>` and `marray<half,{N}>` in the upcoming commit.